### PR TITLE
Require non-empty password to publish password protected posts

### LINKED ISSUE #16344: Post Confirmation Screen: changing to password protected allows not entering password
The new post confirmation function is behind a feature flag so you'll need to test this locally and the AB test overide function in the browser to set `postPublishConfirmation` to `showPublishConfirmation`

#### Steps to reproduce
1. Run `ENABLE_FEATURES=post-editor/delta-post-publish-flow make run`
2. Ensure you're in the `showPublishConfirmation` group
3. Create a new public post click `Publish...`
4. Change status to Password Protected

#### What I expected

Password is required

#### What happened instead

Password is not required

#### Browser / OS version

Chrome OSX

#### Screenshot / Video

password protect

### PR DESCRIPTION
**Pull request by mjuhasz**
Fixes #16344 

This PR makes the password for password-protected publishing required, thus prevents the users from inadvertently publishing public posts by using an empty password.

password-protected-15fps

This is part of a larger epic (See p3Ex-2ri-p2).

To test:
- Checkout this branch
- Run the unit tests: `npm run test-client client/state/posts/test`
- Run this branch with the publish-confirmation feature enabled: `ENABLE_FEATURES=post-editor/delta-post-publish-flow make run`
  - Draft a post
  - Change the visibility to Password Protected under Post Settings --> Status
  - Verify that the password field is empty, it has the focus, no error message is shown yet, and the Publish... button is disabled
  - Click outside the password input field and verify that an error message is shown
  - Type a password and verify that the error message is gone and the Publish... button is enabled
  - Delete the password and verify that after deleting the last character the error message is displayed again and the Publish... button gets disabled
  - Change the visibility back to Public and click Publish...
  - On the confirmation sidebar verify that the Publish! button is enabled, then change the visibility of the post from Public to Password Protected
  - Verify that the behavior is the same as on the editor sidebar in the previous steps as you type/delete password and take away the focus from the field when it's empty
- Run this branch without the publish-confirmation feature enabled: `make run` and verify that the original flow has not been affected (same as master)

### CODE REVIEW

File: client/post-editor/edit-post-status/index.jsx [hunk 1]
```diff
@@ -211,6 +212,10 @@ export class EditPostStatus extends Component {
 			return;
 		}
 
+		if ( 'open' === this.props.confirmationSidebarStatus ) {
```
Comment (mattsherman) [inline client/post-editor/edit-post-status/index.jsx]: Please put a comment here explaining why this is being done.

File: client/post-editor/editor-confirmation-sidebar/index.jsx [hunk 2]
```diff
@@ -28,6 +29,8 @@ class EditorConfirmationSidebar extends React.Component {
 		onPublish: React.PropTypes.func,
 		post: React.PropTypes.object,
 		savedPost: React.PropTypes.object,
+		privatePost: React.PropTypes.bool,
+		privatePostPasswordValid: React.PropTypes.bool,
```
Comment (mattsherman) [inline client/post-editor/editor-confirmation-sidebar/index.jsx]: The naming of these props is a bit confusing. Consider `isPrivatePost` and `isPrivatePostPasswordValid`.

Comment (mjuhasz) [inline client/post-editor/editor-confirmation-sidebar/index.jsx]: Thank you for the review, @mattsherman! I addressed your comments.

**Comment by mattsherman** [comment]
Since merging the latest from master into this branch, this branch now suffers from the intermittent auto publishing issue:

auto scheduling

However, since this PR does not appear to have introduced this issue, I think the best course of action (since this is still behind a feature flag), is to merge this PR and then create a separate PR to address the auto publishing issue solely.

**Pull request merged**